File: flut/flutter/widgets/container.py
from typing import Optional, override

from flut._flut.engine import _flut_pack_value
from flut.dart.ui import Clip
from flut.flutter.widgets.framework import Widget


class Container(Widget):
    _flut_type = "Container"

    def __init__(
        self,
        *,
        key=None,
        padding=None,
        margin=None,
        color=None,
        isAntiAlias: bool = True,
        width: Optional[float] = None,
        height: Optional[float] = None,
        decoration=None,
        foregroundDecoration=None,
        constraints=None,
        alignment=None,
        transform=None,
        transformAlignment=None,
        child: Optional[Widget] = None,
        clipBehavior: Clip = Clip.none,
    ):
        super().__init__(key=key)
        self.padding = padding
        self.margin = margin
        self.color = color
        self.isAntiAlias = isAntiAlias
        self.width = width
        self.height = height
        self.decoration = decoration
        self.foregroundDecoration = foregroundDecoration
        self.constraints = constraints
        self.alignment = alignment
        self.transform = transform
        self.transformAlignment = transformAlignment
        self.child = child
        self.clipBehavior = clipBehavior

    @override
    def _flut_pack(self) -> dict:
        result = self._flut_base_props()
        if self.padding is not None:
            result["padding"] = _flut_pack_value(self.padding)
        if self.margin is not None:
            result["margin"] = _flut_pack_value(self.margin)
        if self.color is not None:
            result["color"] = _flut_pack_value(self.color)
        result["isAntiAlias"] = _flut_pack_value(self.isAntiAlias)
        if self.width is not None:
            result["width"] = _flut_pack_value(self.width)
        if self.height is not None:
            result["height"] = _flut_pack_value(self.height)
        if self.decoration is not None:
            result["decoration"] = _flut_pack_value(self.decoration)
        if self.foregroundDecoration is not None:
            result["foregroundDecoration"] = _flut_pack_value(self.foregroundDecoration)
        if self.constraints is not None:
            result["constraints"] = _flut_pack_value(self.constraints)
        if self.alignment is not None:
            result["alignment"] = _flut_pack_value(self.alignment)
        if self.transform is not None:
            result["transform"] = _flut_pack_value(self.transform)
        if self.transformAlignment is not None:
            result["transformAlignment"] = _flut_pack_value(self.transformAlignment)
        if self.child is not None:
            result["child"] = _flut_pack_value(self.child)
        result["clipBehavior"] = _flut_pack_value(self.clipBehavior)
        return result

File: flut/.flutter/lib/flutter/widgets/container.dart
import 'package:flutter/material.dart';
import 'package:flut/flut/runtime.dart';

class FlutContainer {
  FlutContainer._();

  static Container? flutDecode(FlutRuntime runtime, Map<String, dynamic> data) {
    return Container(
      key: runtime.decodeKey(data),
      padding: runtime.unpackOptionalField<EdgeInsets>(data, 'padding'),
      margin: runtime.unpackOptionalField<EdgeInsets>(data, 'margin'),
      width: runtime.unpackOptionalField<double>(data, 'width'),
      height: runtime.unpackOptionalField<double>(data, 'height'),
      color: runtime.unpackOptionalField<Color>(data, 'color'),
      isAntiAlias: runtime.unpackRequiredField<bool>(data, 'isAntiAlias'),
      decoration: runtime.unpackOptionalField<BoxDecoration>(
        data,
        'decoration',
      ),
      foregroundDecoration: runtime.unpackOptionalField<BoxDecoration>(
        data,
        'foregroundDecoration',
      ),
      constraints: runtime.unpackOptionalField<BoxConstraints>(
        data,
        'constraints',
      ),
      alignment: runtime.unpackOptionalField<Alignment>(data, 'alignment'),
      transform: runtime.unpackOptionalField<Matrix4>(data, 'transform'),
      transformAlignment: runtime.unpackOptionalField<Alignment>(
        data,
        'transformAlignment',
      ),
      clipBehavior: runtime.unpackRequiredField<Clip>(data, 'clipBehavior'),
      child: runtime.unpackOptionalField<Widget>(data, 'child'),
    );
  }
}
